src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1970,17 +1970,26 @@ Connection connectInternal(Properties propsIn,
 
                     if (enclaveAttestationProtocol.equalsIgnoreCase(AttestationProtocol.HGS.toString())) {
                         this.enclaveProvider = new SQLServerVSMEnclaveProvider();
-                    } else {
-                        // If it's a valid Provider & not HGS, then it has to be AAS
+                    } else if (enclaveAttestationProtocol.equalsIgnoreCase(AttestationProtocol.NONE.toString())) {
+                        this.enclaveProvider = new SQLServerNoneEnclaveProvider();
+                    }else {
```

Comment:
hmm seems this is a bad assumption esp now there there are 3 options, should check for AAS and if not, throw error

Reply:
There was a check above for if its a valid protocol, but I agree it would look better to check for each valid protocol and then throw an error for no match. Moved the code block to do this.